tsdb/engine/tsm1/writer_test.go

```diff
@@ -510,3 +511,23 @@ func TestTSMWriter_WriteBlock_Multiple(t *testing.T) {
 		}
 	}
 }
+
+func TestTSMWriter_Write_MaxKey(t *testing.T) {
+	dir := MustTempDir()
+	defer os.RemoveAll(dir)
+	f := MustTempFile(dir)
+	defer f.Close()
+
+	w, err := tsm1.NewTSMWriter(f)
+	if err != nil {
+		t.Fatalf("unexpected error created writer: %v", err)
+	}
+
+	var key string
```

Comment:
What about doing this instead?
```
var key = make([]byte,100000)
for i := range key { key[i] = 'a' }
```

And then just pass `string(key)` to `w.Write()`. I know it's "just a test", but I don't like that we'd be creating over 100000 strings here.